lib/FilesHooks.php

```diff
@@ -1086,6 +1094,7 @@ protected function addNotificationsForUser($user, $subject, $subjectParams, $fil
 		// Add activity to stream
 		if ($streamSetting && (!$selfAction || $this->userSettings->getUserSetting($this->currentUser->getUID(), 'setting', 'self'))) {
 			$this->activityData->send($event);
+			$this->dashboardService->dispatchDashboardEvent($event);
```

Comment:
Yes, the javascript could cycle for new IEvent like when you [open the dashboard page](https://github.com/nextcloud/activity/blob/e27574880e3535a22df34935affbea9c9182a5c1/lib/Widgets/Service/FilesActivityService.php#L68), but the idea is to use the live/push system from Dashboard to update the front-end instead of having the ajax permanently request for updates. I only store the minimum in my own database, and it should be emptied after few hours (or a day).